include/deal.II/meshworker/simple.h

```diff
@@ -211,16 +202,18 @@ namespace MeshWorker
                     const DOFINFO &info2);
     private:
       /**
-       * Assemble a single matrix into #matrix.
+       * Assemble a single matrix <code>M</code> into the element
+       * at <code>index</code> in the vector #matrix.
        */
       void assemble(const FullMatrix<double> &M,
+                    const unsigned int index,
                     const std::vector<types::global_dof_index> &i1,
                     const std::vector<types::global_dof_index> &i2);
 
       /**
        * The global matrix being assembled.
        */
-      SmartPointer<MATRIX,MatrixSimple<MATRIX> > matrix;
+      std::vector<SmartPointer<MATRIX,MatrixSimple<MATRIX> > > matrix;
```

Comment:
It would also be nice to have a specialized return function for the single matrix case - but I do not see a satisfying solution for that...